executor/include/execution/executor/cuda_executor.hpp

```diff
@@ -0,0 +1,47 @@
+//
+// Created by Shrijit Singh on 2020-06-14.
+//
+
+#pragma once
+
+#ifdef CUDA
+#include <cuda_runtime_api.h>
+#endif
+
+#include <execution/executor/base_executor.hpp>
+
+template <typename Interface, typename Cardinality, typename Blocking,
+    typename ProtoAllocator>
+struct cuda_executor;
+
+#ifdef CUDA
+template <> struct execution::executor_available<cuda_executor> : std::true_type {};
+#endif
+
+template <typename Interface,typename Cardinality,typename Blocking, typename ProtoAllocator>
+struct cuda_executor: executor<sse_executor, Interface, Cardinality, Blocking, ProtoAllocator> {
+
+  using shape_type = typename std::array<int, 6>;
+
+  template <typename F, typename... Args>
+  void bulk_execute(F &&f, shape_type shape, Args &&... args) {
+#ifdef CUDA
+    void *kernel_args[] = {&args...};
+    dim3 grid_size(shape[0], shape[1], shape[2]);
+    dim3 block_size(shape[3], shape[4], shape[5]);
+    cudaLaunchKernel((void *)f, grid_size, block_size, kernel_args, 0, 0);
+    cudaDeviceSynchronize();
+#endif
+  }
+
+  auto decay_t() -> decltype(auto) {
+    if constexpr (execution::is_executor_available_t<cuda_executor>()) {
+      return *this;
+    }
+    else
+      return inline_executor<oneway_t, single_t, blocking_t::always_t, ProtoAllocator>{};
+  }
```

Comment:
We should probably move this decay mechanism somewhere else. Something to talk about in the next meeting.

Reply:
I suggested having decay policies as template argument, so that decay mechanism is independent of executor implementation

Reply:
Yeah, as Kunal had suggested that I first look into best_fit executor then come back to decay policies. I have left it as part of v0.1. In later revisions, this can be removed/changed.